Makefile.common

```diff
@@ -245,10 +245,15 @@ common-docker-publish: $(PUBLISH_DOCKER_ARCHS)
 $(PUBLISH_DOCKER_ARCHS): common-docker-publish-%:
 	docker push "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME)-linux-$*:$(DOCKER_IMAGE_TAG)"
 
+PROM_LATEST_VERSION = $(shell cat VERSION)
+DOCKER_IMAGE_TAG_MAJOR = $(firstword $(subst ., ,$(DOCKER_IMAGE_TAG)))
 .PHONY: common-docker-tag-latest $(TAG_DOCKER_ARCHS)
 common-docker-tag-latest: $(TAG_DOCKER_ARCHS)
 $(TAG_DOCKER_ARCHS): common-docker-tag-latest-%:
 	docker tag "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME)-linux-$*:$(DOCKER_IMAGE_TAG)" "$(DOCKER_REPO)/$(DOCKER_IMAGE_NAME)-linux-$*:latest"
+	if [ "$(DOCKER_IMAGE_TAG)" = "v$(PROM_LATEST_VERSION)" ]; then \
```

Comment:
I'm not sure what this is for. When we're tagging latest, we always want to tag the latest major version from the `VERSION` file.

Reply:
#7824 mentions that v2.19.3 was released after v2.20.0. I was trying to consider such cases. I might be missing something, but I thought  https://github.com/prometheus/circleci/blob/7ef6ba93f44340f5a8b5d31e8d04747d0762e660/src/commands/publish_release_images.yml#L41 would cause v2.19.3 to be tagged with `latest` and `v2` after v2.20.0.

Reply:
I think we will need to tackle this in the future. However, I do not think that this solves it.

Reply:
When there is a new release (i.e. v2.20.0), `VERSION` (2.20.0) and `DOCKER_IMAGE_TAG` (v2.20.0) will match and the tag will be updated. If a version like v2.19.3 is published afterwards, `DOCKER_IMAGE_TAG` (v2.19.3) will not match `VERSION` (still 2.20.0 or later) and the `v2` tag will not be changed.

Do you mean this wouldn't solve it as in a problem with the logic, or that another approach would be better? I tried this since #7824 mentions `common-docker-tag-latest`, but there could be other approaches such as a file similar to `VERSION` to point to the desired `v2` or handling it elsewhere in the CI config.

Reply:
No, I think this fine as-is. After reviewing things, we handle release vs release-candidate in the CI config.